lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2373 @@
+      cause problems to call PubNub REST API."
+      - If check should be ignored C(False) should be passed to this parameter.
+    required: false
+    default: true
+'''
+
+EXAMPLES = '''
+# Event handler create example.
+- name: Create single event handler
+  pubnub_blocks:
+    email: '{{ email }}'
+    password: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    name: '{{ block_name }}'
+    event_handlers:
+      -
+        src: '{{ path_to_handler_source }}'
+        name: '{{ handler_name }}'
+        event: 'js-before-publish'
+        channels: '{{ handler_channel }}'
+
+# Change event handler trigger event type.
+- name: Change event handler 'event'
+  pubnub_blocks:
+    email: '{{ email }}'
+    password: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    name: '{{ block_name }}'
+    event_handlers:
+      -
+        name: '{{ handler_name }}'
+        event: 'js-after-publish'
+
+# Stop block and event handlers.
+- name: Stopping block
+  pubnub_blocks:
+    email: '{{ email }}'
+    password: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    name: '{{ block_name }}'
+    state: stop
+
+# Multiple module calls with cached result passing
+- name: Create '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        email: '{{ email }}'
+        password: '{{ password }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+    - name: Add '{{ event_handler_1_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_1_source }}'
+            name: '{{ event_handler_1_name }}'
+            channels: '{{ event_handler_1_channel }}'
+            event: 'js-before-publish'
+    - name: Add '{{ event_handler_2_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_2_source }}'
+            name: '{{ event_handler_2_name }}'
+            channels: '{{ event_handler_2_channel }}'
+            event: 'js-before-publish'
+    - name: Start '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        name: '{{ block_name }}'
+        state: start
+'''
+
+RETURN = '''
+module_cache:
```

Comment:
Instead of module_cache, can you name this something like "auth_cache" or "credentials_cache" since it seems to only contain account information?